src/coordination/coordinator_instance.cpp

```diff
@@ -109,55 +113,14 @@ CoordinatorInstance::~CoordinatorInstance() {
-      thread_pool_.AddTask([this]() { this->ForceResetClusterState(); });
+    if (is_shutting_down_.load(std::memory_order_seq_cst)) {
       return;
     }
-    spdlog::info("Leader changed, starting all replication instances!");
-    auto const instances = raft_state_->GetReplicationInstances();
-    auto replicas = instances | ranges::views::filter(
-                                    [](auto const &instance) { return instance.status == ReplicationRole::REPLICA; });
-
-    std::ranges::for_each(replicas, [this](auto &replica) {
-      spdlog::info("Started pinging replication instance {}", replica.config.instance_name);
-      auto client = std::make_unique<ReplicationInstanceClient>(this, replica.config, client_succ_cb_, client_fail_cb_);
-      repl_instances_.emplace_back(std::move(client), &CoordinatorInstance::ReplicaSuccessCallback,
-                                   &CoordinatorInstance::ReplicaFailCallback);
-    });
-
-    auto main_instances = instances | ranges::views::filter([](auto const &instance) {
-                            return instance.status == ReplicationRole::MAIN;
-                          });
-
-    std::ranges::for_each(main_instances, [this](auto &main_instance) {
-      spdlog::info("Started pinging main instance {}", main_instance.config.instance_name);
-      auto client =
-          std::make_unique<ReplicationInstanceClient>(this, main_instance.config, client_succ_cb_, client_fail_cb_);
-      repl_instances_.emplace_back(std::move(client), &CoordinatorInstance::MainSuccessCallback,
-                                   &CoordinatorInstance::MainFailCallback);
-    });
-
-    // In case we got out of force reset but instances weren't still demoted
-    // we need to apply functions to these instances to demote them
-    auto instances_to_demote =
-        instances | ranges::views::filter([](ReplicationInstanceState const &replication_instance_state) {
-          return replication_instance_state.needs_demote;
-        });
-
-    std::ranges::for_each(instances_to_demote, [this](ReplicationInstanceState const &replication_instance_state) {
-      spdlog::trace("Changing callback for instance {} to demote callback",
-                    replication_instance_state.config.instance_name);
-      auto &instance = FindReplicationInstance(replication_instance_state.config.instance_name);
-      instance.SetCallbacks(&CoordinatorInstance::DemoteSuccessCallback, &CoordinatorInstance::DemoteFailCallback);
-    });
-    std::ranges::for_each(repl_instances_, [](auto &instance) { instance.StartFrequentCheck(); });
-    is_leader_ready_ = true;
+    is_leader_ready_.store(false, std::memory_order_seq_cst);
+    // Thread pool is needed because becoming leader is blocking action, and if we don't succeed to check state of
+    // cluster we will try again and again in same thread, thus blocking progress of nuRAFT leader election
+    thread_pool_.AddTask([this]() { this->VerifyOrCorrectClusterState(); });
```

Comment:
Maybe picky but I still think Reconcile is better name than VerifyOrCorrect, I learnt this term in Kubernetes = reconciliation loop

Reply:
`Reconcile`